armsrc/lfsampling.c

```diff
@@ -305,7 +306,7 @@ uint32_t DoAcquisition(uint8_t decimation, uint8_t bits_per_sample, bool avg, in
 
         // only every 4000th times, in order to save time when collecting samples.
         // interruptible only when logging not yet triggered
-        if ((checked >= 4000) && trigger_hit == false) {
+        if (unlikely(trigger_hit == false && (checked >= 4000))) {
```

Comment:
As per Iceman's comments, use of `unlikely()` is not generally needed.  **_Unless they are needed_**, I support Iceman's position that they should be removed in favor of readability.  This applies to all uses of the `unlikely()` and `likely()` macros....

<details><summary>Details</summary><p/>

In my prior comments during earlier development, there was concern about throughput, and this seemed to be in the time-critical path.  Even then, I couched the comment about GCC extension use with something like, "If timing concerns are critical..." or the like.

The good news is that you seem to have moved away from the very difficult "every processor cycle matters" development.  Well done!   You are now in the happier zone where profiling, hot code paths, etc. can be left behind as bad memories.

BTW ... you did a great job migrating the GCC extensions for this, and wrapping them into clear, easy-to-use macros.  Well done, even if pulled for the final PR!

</details>

Reply:
I see. I've tested the code without `unlikely()` and the real-time sampling also works. I've removed them.
It's good to know these GCC extensions. Thank you for your guidance!